lib/register.php

```diff
@@ -360,6 +360,32 @@ function gutenberg_register_rest_routes() {
 add_action( 'rest_api_init', 'gutenberg_register_rest_routes' );
 
 
+function gutenberg_get_post_revisions( $post ) {
+	$revisions = wp_get_post_revisions( $post['id'] );
+	$revisions_count = count( $revisions );
+	if ( $revisions_count === 0 ) {
+		return null;
+	}
+
+	$last_revision = array_shift( $revisions );
+
+	return array(
+		'count' => $revisions_count,
+		'last_id' => $last_revision->ID
+	);
+}
+
+function gutenberg_register_post_revisions() {
+	register_rest_field( get_post_types( '', 'names' ),
+		'revisions',
+		array(
+			'get_callback'  => 'gutenberg_get_post_revisions',
+		)
+	);
+}
+add_action( 'rest_api_init', 'gutenberg_register_post_revisions' );
```

Comment:
Just wondering if it's a good idea to add extra fields to this endpoint. This won't be specific to Gutenberg.

Reply:
I'm learning WP hooks and APIs. I'm open to suggestions how to implement it better. This change makes loading Gutenberg with tons of revisions comparable to the Classic editor.

Reply:
I still think a separate endpoint is better, do you know why the previously used endpoint was slow? Maybe we should create a custom endpoint? Also, I'm not 100% confident when it comes to the REST API, others might have a better input.

Reply:
@youknowriad: Loading all the revisions was running pretty much all of the code that would be run if the posts were being loaded to display on the front end. `the_content` was the biggest problem, but everything added up.

Reply:
I'm okay with the field being here for now. I was unfortunately waylaid for the past week or so, so I haven't managed to focus on the REST API problem, but I'm hoping to get back to it later this week. I imagine there'll be a lot of re-shuffling, so I wouldn't get too hung up on where this is for now.

Reply:
Yes, I'm aware of that, I'm just saying we may want to load the "last_revision_id" and "revision_count" in a separate endpoint instead of adding these as extra fields to the default POST endpoint. Because this change will change the result for everything and not only Gutenberg.